src/hpr_routing_config.erl

```diff
@@ -0,0 +1,257 @@
+            %%         ({{NetID0, Start, End}, Route}) when
+            %%             NetID0 == NetID andalso Start =< DevAddr andalso DevAddr =< End
+            %%         ->
+            %%             Route
+            %%     end
+            %% ),
+            MS = [
+                {
+                    {{'$1', '$2', '$3'}, '$4'},
+                    [
+                        {'andalso', {'==', '$1', NetID},
+                            {'andalso', {'==', '$2', 0}, {'==', '$3', 0}}}
+                    ],
+                    ['$4']
+                },
+                {
+                    {{'$1', '$2', '$3'}, '$4'},
+                    [
+                        {'andalso', {'==', '$1', NetID},
+                            {'andalso', {'=<', '$2', DevAddr}, {'=<', DevAddr, '$3'}}}
+                    ],
+                    ['$4']
+                }
+            ],
+            ets:select(?DEVADDRS_ETS, MS);
+        _Err ->
+            []
+    end.
+
+%% ------------------------------------------------------------------
+%% gen_server Function Definitions
+%% ------------------------------------------------------------------
+init(#{base_dir := BaseDir} = _Args) ->
+    File = filename:join(BaseDir, erlang:atom_to_list(?DETS)),
+    ok = filelib:ensure_dir(File),
+    lager:info("init with dets=~s", [File]),
+    {ok, ?DETS} =
+        dets:open_file(?DETS, [{file, File}]),
+    _ = ets:new(?EUIS_ETS, [
+        public,
+        named_table,
+        bag,
+        {read_concurrency, true}
+    ]),
+    _ = ets:new(?DEVADDRS_ETS, [
+        public,
+        named_table,
+        bag,
+        {read_concurrency, true}
+    ]),
+    ok = init_ets(),
+    {ok, #state{}, {continue, ?INIT_ASYNC}}.
+
+handle_continue(?INIT_ASYNC, State) ->
+    %% Connect to Config Service to get updates
+    {noreply, State};
+handle_continue(_Msg, State) ->
+    {noreply, State}.
+
+handle_call(_Msg, _From, State) ->
+    {reply, ok, State}.
+
+handle_cast(_Msg, State) ->
+    {noreply, State}.
+
+handle_info(_Msg, State) ->
+    {noreply, State}.
+
+code_change(_OldVsn, State, _Extra) ->
+    {ok, State}.
+
+terminate(_Reason, _State) ->
+    _ = dets:close(?DETS),
+    _ = ets:delete(?EUIS_ETS),
+    _ = ets:delete(?DEVADDRS_ETS),
+    ok.
+
+%% ------------------------------------------------------------------
+%% Internal Function Definitions
+%% ------------------------------------------------------------------
+
+-spec init_ets() -> ok.
+init_ets() ->
+    {EUIRoutes, DevAddrRoutes} = dets:foldl(
+        fun({_OUI, Route}, {EUIRoutesAcc, DevAddrRoutesAcc}) ->
+            EUIs = hpr_routing_config_route:euis(Route),
+            EUIRoutes = [{{AppEUI, DevEUI}, Route} || {AppEUI, DevEUI} <- EUIs],
+            NetID = hpr_routing_config_route:net_id(Route),
+            DevAddrRangeRoutes =
+                case hpr_routing_config_route:devaddr_ranges(Route) of
+                    [] ->
+                        [{{NetID, 0, 0}, Route}];
```

Comment:
When there's no range, we can insert min and max values.
ref: https://github.com/helium/packet-purchaser/blob/master/src/pp_config_v2.erl#L108

This allows the spec when looking up DevAddr ranges to only need to do 2 comparisons instead of 4. 

When we look up DevAddrs by range, we already have the NetID, so the possibility of sending data to the wrong range is very low.